src/hedera/hederaEngine.js

```diff
@@ -0,0 +1,404 @@
+      networkFee: '0', // networkFee
+      ourReceiveAddresses: [], // ourReceiveAddresses
+      signedTx: '', // signedTx
+      otherParams: {
+        fromAddress: this.walletLocalData.publicKey,
+        toAddress: publicAddress,
+        transferTx: base64.encode(transferTx.toBytes())
+      }
+    }
+
+    this.log(
+      `${nativeAmount} ${this.walletLocalData.publicKey} -> ${publicAddress}`
+    )
+    return edgeTransaction
+  }
+
+  async signTx(edgeTransaction: EdgeTransaction): Promise<EdgeTransaction> {
+    if (
+      !edgeTransaction.otherParams ||
+      !edgeTransaction.otherParams.transferTx
+    ) {
+      throw new Error('missing otherParam transferTx')
+    }
+
+    const { hederaPrivateKey } = this.walletInfo.keys
+
+    if (!hederaPrivateKey) {
+      throw new Error('missing hederaPrivateKey in walletInfo')
+    }
+
+    const transferTx = hedera.Transaction.fromBytes(
+      this.client,
+      base64.decode(edgeTransaction.otherParams.transferTx)
+    )
+    await transferTx.sign(hedera.Ed25519PrivateKey.fromString(hederaPrivateKey))
+
+    return {
+      ...edgeTransaction,
+      otherParams: {
+        ...edgeTransaction.otherParams,
+        signedTx: base64.encode(transferTx.toBytes())
+      }
+    }
+  }
+
+  async broadcastTx(
+    edgeTransaction: EdgeTransaction
+  ): Promise<EdgeTransaction> {
+    if (!edgeTransaction.otherParams || !edgeTransaction.otherParams.signedTx) {
+      throw new Error('InvalidTransactionParams')
+    }
+
+    try {
+      const { signedTx } = edgeTransaction.otherParams
+      const txn = hedera.Transaction.fromBytes(base64.decode(signedTx))
+      await txn.execute(this.client)
+      await txn.getReceipt(this.client)
+    } catch (e) {
+      this.log(e)
+      throw e
+    }
+    // must be > 0 to not show "Synchronizing"
+    edgeTransaction.blockHeight = 1
+    return edgeTransaction
+  }
+
+  async getAccountId(): Promise<?hedera.AccountId> {
+    this.log('hederaEngine/getAccountId', {
+      publicKey: this.walletInfo.keys.publicKey
+    })
+
+    const apiServer = this.currencyInfo.defaultSettings.otherSettings
+      .kabutoApiServers[0]
+
+    const response = await (
+      await this.io.fetch(
+        `${apiServer}/v1/account?key=${this.walletInfo.keys.publicKey}`
+      )
+    ).json()
+
+    if (response.accounts && response.accounts.length > 0) {
+      this.log('returned accounts', response.accounts)
+      return new hedera.AccountId(response.accounts[0].id)
+    } else {
+      return null
+    }
+  }
+
+  getFreshAddress(options: object): EdgeFreshAddress {
+    // same as EOS, return account ID if we have one, otherwise return an empty string
+    if (this.walletLocalData.otherData.hederaAccount) {
+      return { publicAddress: this.walletLocalData.otherData.hederaAccount }
+    } else {
+      return {
+        publicAddress: '',
+        publicKey: this.walletInfo.keys.publicKey
+      }
+    }
+  }
+
+  getBlockHeight(): number {
+    return 2
```

Comment:
Even returning this value, the wallet still shows as synchronizing for some reason.

Reply:
Needs to call `this.currencyEngineCallbacks.onAddressesChecked` with 1 when syncing is done.